### PR TITLE
Longshot recipe

### PR DESCRIPTION
**Pull request by pjedge**
:information_source:
Bioconda has finished the [GCC7 migration](https://github.com/bioconda/bioconda-recipes/issues/13578). If you are dealing with C/C++ or Python package it can be that you need to rebuild other dependent packages. [Bioconda utils - update-pinning](https://bioconda.github.io/updating.html#updating-recipes-for-a-pinning-change) will assist you with that. If you have any questions please use [issue 13578](https://github.com/bioconda/bioconda-recipes/issues/13578).

----------------

* [ X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ X] This PR adds a new recipe.
* [ X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [] This PR updates an existing recipe.
* [] This PR does something else (explain below).

### CODE REVIEW

**Comment by pjedge** [comment]
If someone could review this recipe for me, that'd be great. @bioconda/core
It's written in Rust, and the recipe is based on the rust-bio-tools recipe with minimal modification.

**Comment by dpryan79** [comment]
@johanneskoester I think you're the rust expert among us, can you have a look?

**Comment by pjedge** [comment]
Hi, can someone please review this recipe for me?

File: recipes/longshot/build.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# this build script is taken from the rust-bio-tools recipe
+# https://github.com/bioconda/bioconda-recipes/blob/master/recipes/rust-bio-tools/build.sh
+
+# taken from yacrd recipe, see: https://github.com/bioconda/bioconda-recipes/blob/2b02c3db6400499d910bc5f297d23cb20c9db4f8/recipes/yacrd/build.sh
+if [ "$(uname)" == "Darwin" ]; then
+
+    # apparently the HOME variable isn't set correctly, and circle ci output indicates the following as the home directory
+    export HOME="/Users/distiller"
+
+    # according to https://github.com/rust-lang/cargo/issues/2422#issuecomment-198458960 removing circle ci default configuration solves cargo trouble downloading crates
```
Comment (dpryan79) [inline recipes/longshot/build.sh]: I'll show my ignorance of rust here (I really should start a project using it), but what dependencies or other crates does this recipe require? Ideally we'd make those conda packages too (assuming that makes any sense at all for rust).

File: recipes/longshot/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    - make
+    - {{ compiler('c') }}  # [not osx]
+    - {{ compiler('cxx') }}  # [not osx]
+    - clang_osx-64 # [osx]
+    - clangxx_osx-64 # [osx]
+    - pkg-config  # [osx]
+  host:
+    - zlib
+    - xz
+    - clangdev
```
Comment (dpryan79) [inline recipes/longshot/meta.yaml]: Do you really need this? If so, even under Linux?

File: recipes/longshot/build.sh (hunk 1 above)
Comment (pjedge) [reply recipes/longshot/build.sh]: The recipe requires a whole host of crates, but they are managed automatically by cargo. I don't think it's feasible to have them be managed by conda instead. I based this recipe off Johannes' rust-bio-tools recipe (very little is changed, it basically specifies some non-rust dependencies and calls ```cargo install```).

File: recipes/longshot/meta.yaml (hunk 2 above)
Comment (pjedge) [reply recipes/longshot/meta.yaml]: Yeah they are definitely all required. If I remember correctly, all three are rust-htslib dependencies.

**Comment by dpryan79** [comment]
Thanks, looks fine to me then (not that that's worth terribly much, given my obvious lack of familiarity with rust).

**Comment by pjedge** [comment]
Thanks. I think I may publish a second release in the coming days (recipe will be the same except version number bump), and if I do that can I merge it myself?

**Comment by dpryan79** [comment]
Unless someone else from the core team objects before then I would take that at tacit approval.

File: recipes/longshot/build.sh (hunk 1 above)
Comment (johanneskoester) [reply recipes/longshot/build.sh]: Rust statically compiles pure-rust dependencies stuff into the binary. This is all fine. Those deps usually very small.

File: recipes/longshot/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  sha256: ec8ebfb6007d42c6952bac6843e903f899f33d6c0fadbd976cc272d8d7f4b279
+
+requirements:
+  build:
+    - rust >=1.32
+    - make
+    - {{ compiler('c') }}  # [not osx]
+    - {{ compiler('cxx') }}  # [not osx]
+    - clang_osx-64 # [osx]
+    - clangxx_osx-64 # [osx]
```
Comment (johanneskoester) [inline recipes/longshot/meta.yaml]: Can't we use the compiler function here as well? I think the default compiler of macos is clang anyway.

Comment (pjedge) [reply recipes/longshot/meta.yaml]: I removed those and left just the compiler function -- passed tests. thanks!

**Comment by bgruening** [comment]
Thanks @pjedge!

**Comment by pjedge** [comment]
Thanks for all your help! I tested installation on a centos machine and it worked great! However, I asked a friend to try installation on their macbook and it failed with the following error:
```
PackageNotFoundError: Packages missing in current channels:
 - longshot -> libcxx >=4.0.1
```
Any idea what I can do about this?

**Comment by dpryan79** [comment]
@pjedge Make sure they still have `defaults` in their list of channels (that's where that package should be coming from).

**Pull request merged**